### PR TITLE
Review comments:

### PR DESCRIPTION
**Pull request by Liamsi**
Some changes to #100 in separate PR to s.t. other PRs (#104, #105 etc) don't need to rebase / merge 

* [ ] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGES.md

### CODE REVIEW

File: tendermint/src/lite/verifier.rs [hunk 1]
```diff
@@ -194,7 +195,9 @@ where
         now,
     ) {
         Ok(_) => {
-            store.add(untrusted_header)?;
+            let untrusted_next_vals = req.validator_set(untrusted_header.header().height())?;
```
Comment (ebuchman) [inline tendermint/src/lite/verifier.rs]: Shouldn't this be height+1 ?

File: tendermint/src/lite/verifier.rs [hunk 2]
```diff
@@ -241,8 +245,10 @@ where
         req,
         store,
     )?;
-    // Add to trusted headers:
-    store.add(untrusted_header)?;
+    // Add header (and corresponding next validators) to trusted state store:
+    let untrusted_next_vals = req.validator_set(untrusted_header.header().height())?;
```
Comment (ebuchman) [inline tendermint/src/lite/verifier.rs]: same, +1?

File: tendermint/src/lite/verifier.rs [hunk 3]
```diff
@@ -261,29 +267,24 @@ pub fn verify_header<TS, SH, VS, L, S, R>(
...
     TS: TrustedState<LastHeader = SH, ValidatorSet = VS>,
     L: TrustThreshold,
-    S: Store<SignedHeader = SH>,
+    S: Store<TrustedState = TS>,
     R: Requester<SignedHeader = SH, ValidatorSet = VS>,
     VS: ValidatorSet,
     SH: SignedHeader,
 {
     // Check if we already trusted a header at the given height and it didn't expire:
     if let Ok(sh2) = store.get(height) {
```
Comment (ebuchman) [inline tendermint/src/lite/verifier.rs]: We should probably change the `sh` to `ts` wherever signed header became trusted state.

**Comment by Liamsi** [comment]
Merging this as it's probably better to have only one open PR regarding the light client instead of several.

**Pull request merged**